### DISCUSSION TITLE
charge_history API's response is empty JSON only.

### DISCUSSION THREAD

**Discussion by earshttp** [timdorr/tesla-api#592] [Q&A] [answered]
I called this endpoint by POST method with authorization header.

POST https://owner-api.teslamotors.com/api/1/vehicles/xxxxxxxxxx/charge_history
Authorization: Bearer xxxxxxxxx

charge_history API is succeed but responsed below.

```
{
"response": "{}\n"
}
```

I can see charging historical datas between 30 days in My Tesla app, so I'm wonrdering why.

Is there any params to get valid response or this API is not available yet?

**Comment by earshttp** [comment, marked as answer]
I solved myself.
I added to request header.

```
x-tesla-user-agent: TeslaApp/4.9.2
```

**Reply by rawmean** [reply]
Do we need to specify a timezone for this endpoint? It seems that the charging info presented in the Tesla app is ahead by one day compared to the info provided in the response to this http endpoint by one day when the user is in the UK.

**Reply by earshttp** [reply]
@rawmean
So far, no such parameter has been found.
After getting the response, you may need to process it yourself.
As far as the value of timestamp is seen, the date and time based on Los Angeles (UTC -7) is returned.
